src/darwin/frida-helper-backend-glue.m

```diff
@@ -882,51 +1166,30 @@ static gboolean frida_agent_context_init_functions (FridaAgentContext * self, Gu
   vm_deallocate (self_task, (vm_address_t) threads, thread_count * sizeof (thread_t));
   threads = NULL;
 
-  ret = thread_get_state (child_thread, state_flavor, (thread_state_t) &state, &state_count);
-  CHECK_MACH_RESULT (ret, ==, KERN_SUCCESS, "thread_get_state");
-
-#ifdef HAVE_I386
-  dyld_start = (instance->cpu_type == GUM_CPU_AMD64) ? state.uts.ts64.__rip : state.uts.ts32.__eip;
-#else
-  dyld_start = (instance->cpu_type == GUM_CPU_ARM64) ? state.ts_64.__pc : state.ts_32.__pc;
-#endif
-
-  dyld_header = 0;
-  dyld_granularity = 4096;
-  for (dyld_chunk = (dyld_start & (dyld_granularity - 1)) == 0 ? (dyld_start - dyld_granularity) : (dyld_start & ~(dyld_granularity - 1));
-      dyld_header == 0;
-      dyld_chunk -= dyld_granularity)
-  {
-    guint32 * magic;
+  if (cached_address_for_breakpoint[instance->cpu_type] == 0) {
+    gum_darwin_enumerate_modules (task, find_libSystem, &libSystem_header);
```

Comment:
yeah, it's taking the slow path. it takes between around 40 and 80 milliseconds on my tests (iPhone 6 with iOS 10.2 in this case, measured using `mach_absolute_time`). This is partially compensated by caching the address, so in iDevice scenario in which the server is running, this delay is only on the first launch (per target CPU type).